File: app/src/main/java/com/mutant/godutch/ViewPagerAdapter.java
package com.mutant.godutch;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by evanfang102 on 2017/3/30.
 */

public class ViewPagerAdapter extends FragmentPagerAdapter {

    List<Fragment> mFragments = new ArrayList<>();

    public ViewPagerAdapter(FragmentManager manager) {
        super(manager);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    public void addFragment(Fragment fragment) {
        mFragments.add(fragment);
    }
}

File: app/src/main/java/com/mutant/godutch/ViewPagerAdapter.kt
package com.mutant.godutch

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.*

/**
 * Created by evanfang102 on 2017/3/30.
 */

class ViewPagerAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {

    internal var mFragments: MutableList<Fragment> = ArrayList()

    override fun getCount(): Int {
        return mFragments.size
    }

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    fun addFragment(fragment: Fragment) {
        mFragments.add(fragment)
    }
}
